M365/Remove-MailboxExtendedProperty.ps1

```diff
@@ -0,0 +1,72 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#Requires -Modules @{ ModuleName="ExchangeOnlineManagement"; RequiredVersion="3.4.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Users"; RequiredVersion="2.24.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Mail"; RequiredVersion="2.24.0" }
+
+<#
+.SYNOPSIS
+    Removes the extended property (aka named property) the message.
+
+.DESCRIPTION
+    Takes as input the result from Search-MailboxExtendedProperty.ps1 and removes the extended property from the message.
+
+.PARAMETER MessagesWithExtendedProperty
+    A list of mailbox items and their extended property, that are to be removed.
+
+.EXAMPLE
+    $mailboxExtendedProperty = Get-MailboxExtendedProperty -Identity fred@contoso.com | Where-Object { $_.PropertyName -like '*Some Pattern*' }
+    $messagesWithExtendedProperty = .\Search-MailboxExtendedProperty.ps1 -MailboxExtendedProperty $mailboxExtendedProperty
+    .\Remove-MailboxExtendedProperty.ps1 -MessagesWithExtendedProperty $messagesWithExtendedProperty
+
+#>
+[CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'High')]
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
+    [ValidateScript({
+            if ($_.GetType().FullName -eq 'System.Management.Automation.PSCustomObject' -or $_.GetType().FullName -eq 'System.Object[]') {
+                $true
+            } else {
+                throw "The parameter MailboxExtendedProperty doesn't appear to be the result from running 'Search-MailboxExtendedProperty'."
+            }
+        })]
+    $MessagesWithExtendedProperty
+)
+
+process {
+    # Get the current Microsoft Graph context
```

Comment:
You can add the shared `GenericScriptUpdate.ps1` helper functions to the script. This will add auto-update capabilities in case that we release a newer version of the script.

Reply:
If we don't think the scripts are going to have a lot of repeat traffic, we might not need to have the update script feature. @cboonham I would leave that up to your discretion.